website/components/AppWrapper.tsx

```diff
@@ -56,20 +70,47 @@ export default function AppWrapper(inputProps: AppWrapperProps): ReactElement {
     // On load, fetch parameters from the URL and location state, then merge.
     const locationArgs = location.state as AppDataProps;
     getArgsFromParams(searchParams).then(({ args: urlArgs, viewerSettings: urlViewerSettings }) => {
-      setViewerArgs({ ...DEFAULT_APP_PROPS, ...locationArgs, ...urlArgs });
+      setViewerArgs({ ...DEFAULT_APP_PROPS, ...urlArgs, ...locationArgs });
       setViewerSettings({ ...DEFAULT_VIEWER_SETTINGS, ...urlViewerSettings });
     });
   }, []);
 
+  // TODO: Disabled for now, since it only makes sense for Zarr/OME-tiff URLs. Checking for
+  // validity may be more complex. (Also, we could add a callback to `ImageViewerApp` for successful
+  // loading and only save the URL then.)
+  //
+  // Save recent zarr data urls
+  // useEffect(() => {
+  //   if (typeof viewerArgs.imageUrl === "string" && isValidZarrUrl(viewerArgs.imageUrl)) {
+  //     // TODO: Handle case where there are multiple URLs?
+  //     // TODO: Save ALL AppProps instead of only the URL? Ignore/handle rawData?
+  //     addRecentDataUrl({ url: viewerArgs.imageUrl as string, label: viewerArgs.imageUrl as string });
+  //   }
+  // }, [viewerArgs]);
+
+  const onLoad = (appProps: AppDataProps): void => {
+    // Force a page reload. This prevents a bug where a desync in the number of channels
+    // in the viewer can cause a crash. The root cause is React immediately forcing a
+    // re-render every time `setState` is called in an async function.
+    const url = appProps.imageUrl;
+    if (url instanceof Array) {
```

Comment:
Same suggestion, also optional:

```suggestion
    if (Array.isArray(url)) {
```